compiler/noirc_evaluator/src/acir/acir_context/mod.rs

```diff
@@ -981,7 +981,10 @@ impl<F: AcirField, B: BlackBoxFunctionSolver<F>> AcirContext<F, B> {
 
         // Optimization when rhs is const and fits within a u128
         let rhs_expr = self.var_to_expression(rhs)?;
-        if rhs_expr.is_const() && rhs_expr.q_c.num_bits() <= 128 {
+        // Do not attempt this optimization when the q_c is zero as otherwise
+        // we will compute an rhs offset of zero and ultimately lay down a range constrain of zero bits
+        // which will always fail.
```

Comment:
Instead of avoiding this when the range constraint has bit-size 0, I'd rather fix the range constraint so that it does not fail with 0: if the bit size is 0, we can simply AssertZero() the variable (to be 0) in the range constraint.

Reply:
We've defined elsewhere in the codebase that a value of 0 requires 1 bit to store it so there's scope for introducing new issues due to this.

Reply:
I'll capture this in an issue

Reply:
https://github.com/noir-lang/noir/issues/8251